api_app/api/routes/workspaces.py

```diff
@@ -229,6 +229,16 @@ async def create_workspace_service(response: Response, workspace_service_input:
         logging.error(f"User not authorized to use template: {e}")
         raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
 
+    # if template has address_space get an address space
+    if resource_template.properties.get("address_space"):
+        # check workspace has address_spaces property
+        if not workspace.properties.get("address_spaces"):
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=strings.WORKSPACE_DOES_NOT_HAVE_ADDRESS_SPACES_PROPERTY)
```

Comment:
Is this really about the current request?
Since we expect every workspace to have an address space anyway should it be an HTTP 500?

Reply:
I think it is the current request, what we are doing is saying if this is a workspace without an `address_spaces` (not `address_space`) property it cannot succeed. At present there are TREs deployed with workspaces, and workspace templates without the `address_spaces` property.